src/pages/products-page/ProductPage.jsx

```diff
@@ -1,5 +1,162 @@
+import "./product.css";
+import { useState, useEffect,useReducer } from "react";
+import axios from "axios";
+import {ProductList} from "../../components/ProductList/ProductList"
+
 export default function ProductPage(){
+        const [products,setProducts]=useState([])
+        useEffect(() => {
+            const fetchProducts = async () => {
+              const res = await axios.get("/api/products");
+              setProducts(res.data.products);
+            };
+            fetchProducts();
+          }, []);
+          const ProductReducerFunction=(state,action)=>{
+            switch(action.type){
+                case "sort":
```

Comment:
this "sort" can be moved to separate constants file and that can be used in other places